core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -63,12 +68,11 @@ class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
       this.subscriptions = subscriptions;
   }
 
-  def addSubscription(subscription: Subscription): Unit = {
+  def addSubscription(subscription: Subscription): Future[Any] = {
```

Comment:
this can still return a Unit as it was, this method is only called from the Actor's `receive` method and none outside will call this method directly.

Reply:
btw: all of the methods can be made private in this actor